src/jit/optcse.cpp

```diff
@@ -1239,6 +1403,54 @@ void Compiler::optValnumCSE_Availablity()
                         desc->csdUseWtCnt += stmw;
                     }
                 }
+
+                // In order to determine if a CSE is live across a call, we model availablity using two bits
+                // and kill the cseBit2 for each CSE whenever we see a GT_CALL
+                //
+                if (tree->OperGet() == GT_CALL)
+                {
+                    // Check for the common case of an already empty available_cses set
+                    // and thus nothing needs to be killed
+                    //
+                    if (!(BitVecOps::IsEmpty(cseTraits, available_cses)))
+                    {
+                        // We can have a GT_CALL that produces a CSE,
+                        // (i.e. HELPER.CORINFO_HELP_GETSHARED_*STATIC_BASE or CORINFO_HELP_TYPEHANDLE_TO_RUNTIMETYPE,
+                        // etc..) Either isUse or isDef will be set to true in this case.
+
+                        // For a Def we will stil have a call, but it makes one CSE available so don't kill it
+                        // but still kill all the other CSEs
+                        //
+                        if (isDef)
+                        {
+
+                            EXPSET_TP call_killset = BitVecOps::UninitVal();
+                            BitVecOps::Assign(cseTraits, call_killset, call_kills_all_cses);
```

Comment:
This will always allocate a bitset when the number of CSEs exceeds 32. You could hoist this bitset out of the loop to avoid this. But this acrobatics seems unnecessary, you could just save `cseBit2` bit from `available_cses` to a variable and restore it after doing the intersection.